File: client/signalrChatClient/src/app/models/message.ts
export class Message{
    text: string;
    sendDate: Date;
    sourceNickName: string;
    destNickName: string;
    constructor(item: any = null){
        this.text = item && item.text || null;
        this.sourceNickName = item && item.sourceNickName || null;
        this.destNickName = item && item.destNickName || null;
        this.sendDate = item && item.sendDate || Date.now;
    }
}

File: server/signalrChat/signalrChat/Models/Message.cs
using System;

namespace signalrChat.Models
{
    /// <summary>
    /// Модель сообщения
    /// </summary>
    public class Message
    {
        /// <summary>
        /// Текст сообщения
        /// </summary>
        public string Text { get; set; }

        /// <summary>
        /// Когда отправлена
        /// </summary>
        public DateTimeOffset SendDate { get; set; }

        /// <summary>
        /// ID отправителя
        /// </summary>
        public string SourceNickName { get; set; }

        /// <summary>
        /// Статус
        /// </summary>
        public MessageStates State { get; set; }

        /// <summary>
        /// Ник получателя(если нет то в общий чат)
        /// </summary>
        public string DestNickName { get; set; }
    }

    public enum MessageStates: int
    {
        /// <summary>
        /// Отправлена
        /// </summary>
        Sent = 0,

        /// <summary>
        /// Получена
        /// </summary>
        Received = 1,

        /// <summary>
        /// Ошибка
        /// </summary>
        Error = 255,
    }
}
